### PR TITLE
Users/achalla/results directory fix

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Fix for relative result directory in settings file.
Testing : manual and unit tests

### CODE REVIEW

**Comment by AshwiniChalla** [comment]
refactored mock response utilities.

**Comment by nigurr** [comment]
Squash all commits into one

File: Tasks/VsTest/vstest.ts [hunk 1]
```diff
@@ -472,7 +472,16 @@ function getTestResultsDirectory(settingsFile: string, defaultResultsDirectory:
             parser.parseString(xmlContents, function (err, result) {
                 if (!err && result.RunSettings && result.RunSettings.RunConfiguration && result.RunSettings.RunConfiguration[0] &&
                     result.RunSettings.RunConfiguration[0].ResultsDirectory && result.RunSettings.RunConfiguration[0].ResultsDirectory[0].length > 0) {
-                    defer.resolve(result.RunSettings.RunConfiguration[0].ResultsDirectory[0]);
+                    var resultDirectory = result.RunSettings.RunConfiguration[0].ResultsDirectory[0];
+                    resultDirectory = resultDirectory.trim();
+                    
+                    if (resultDirectory) {
```
Comment (nigurr) [inline Tasks/VsTest/vstest.ts]: does this has to be async function?

**Comment by nigurr** [comment]
Assuming you have run all the UTs in Win/*nix machines. 
Fix the review comments. approved

File: Tasks/VsTest/vstest.ts (hunk 1 above)
Comment (AshwiniChalla) [inline Tasks/VsTest/vstest.ts]: yes. read function is async

**Comment by AshwiniChalla** [comment]
raised different pr: https://github.com/Microsoft/vsts-tasks/pull/1989

**Pull request closed without merging**